### ISSUE TITLE
get_model_filtered_variables operation not taking into account model Usage Restrictions

### ISSUE THREAD

**Issue by mhungria** [cdisc-org/cdisc-rules-engine#1816]
### Standard

CDISC SDTM/SDTMIG

### Reference Rule ID(s)

CG0219

### Conformance Rule ID(s) (if published in CORE)

CORE-000575

### JIRA Ticket

https://jira.cdisc.org/browse/CORERULES-183?jql=text%20~%20%22cg0219%22

### CLI Command Used (if applicable)

python core.py -s SDTM -v 3.4 -d /path/to/datasets

### Rule YAML

```YAML
Operations:
  - id: $dataset_variables
    operator: get_column_order_from_dataset
  - id: $timing_variables
    key_name: role
    key_value: Timing
    operator: get_model_filtered_variables
```

### Attach any sample test data file(s)

As in the previously published rules, now under revision for v1.0

### Attach any output report and/or log file(s)

"['VISITNUM', 'VISIT', 'VISITDY', 'TAETORD', 'EPOCH', 'RPHASE', 'RPPLDY', 'RPPLSTDY', 'RPPLENDY', 'LBDTC', **'LBSTDTC',** 'LBENDTC', 'LBDY', **'LBSTDY',** 'LBENDY', 'LBNOMDY', 'LBNOMLBL', 'LBRPDY', 'LBRPSTDY', 'LBRPENDY', 'LBXDY', **'LBXSTDY'**, 'LBXENDY', 'LBCHDY', **'LBCHSTDY',** 'LBCHENDY', 'LBDUR', 'LBTPT', 'LBTPTNUM', 'LBELTM', 'LBTPTREF', 'LBRFTDTC', 'LBSTRF', 'LBENRF', 'LBEVLINT', 'LBEVINTX', 'LBSTRTPT', 'LBSTTPT', 'LBENRTPT', 'LBENTPT', 'MIDS', 'RELMIDS', 'MIDSDTC', 'LBSTINT', 'LBENINT', 'LBDETECT', 'LBPTFL', 'LBPDUR']"

### Expected output

"['VISITNUM', 'VISIT', 'VISITDY', 'TAETORD', 'EPOCH', 'RPHASE', 'RPPLDY', 'RPPLSTDY', 'RPPLENDY', 'LBDTC',  'LBENDTC',  'LBENDY', 'LBNOMDY', 'LBNOMLBL', 'LBRPDY', 'LBRPSTDY', 'LBRPENDY', 'LBXDY', 'LBXENDY', 'LBCHDY', 'LBCHENDY', 'LBDUR', 'LBTPT', 'LBTPTNUM', 'LBELTM', 'LBTPTREF', 'LBRFTDTC', 'LBSTRF', 'LBENRF', 'LBEVLINT', 'LBEVINTX', 'LBSTRTPT', 'LBSTTPT', 'LBENRTPT', 'LBENTPT', 'MIDS', 'RELMIDS', 'MIDSDTC', 'LBSTINT', 'LBENINT', 'LBDETECT', 'LBPTFL', 'LBPDUR']"

### Any Additional Information

<img width="1243" height="903" alt="Image" src="https://github.com/user-attachments/assets/bb6c39e2-d89b-4f76-bcbb-986d2063dac7" />
